src/components/FileUploadButton.tsx

```diff
@@ -27,18 +28,8 @@ export const FileUploadButton: React.FC<FileUploadButtonProps> = ({
       <label>
         <input
           type="file"
-          ref={inputRef}
           className="rfu-file-input"
-          onChange={(event) => {
-            const files = event.currentTarget.files;
-            if (files) {
-              handleFiles(files);
-            }
-            if (inputRef.current !== null) {
-              inputRef.current.value = '';
-              inputRef.current.blur();
-            }
-          }}
+          onChange={handleFileChange(handleFiles, resetOnChange)}
```

Comment:
```suggestion
          onChange={() => handleFileChange(handleFiles, resetOnChange)}
```

You sure this is right? You are calling the function on every render ... not only on chance

Reply:
Yes, `handleFileChange` is just a closure that takes into consideration actual user handler (in this case `handleFiles`) and `resetOnChange` property and returns function that handles `onChange` event. I could memoize it if you think it'd have impact on performance.

Reply:
Ahh ok, I missed the part where you return the function from handleFileChange, my bad. Memoization is not necessary. But handleFileChange doesn't exactly handle the file change event. It only gives you handler. You can consider following change, which avoids calling that function on every render and does exactly what it says:

```jsx
// utils.js
export const handleFileChange = (
  files: WhateverType,
  resetOnChange: boolean = false,
  handler?: (files: Array<File>) => void, // last since its optional
) => {
  if (!files) return;

  try {
    handler?.(Array.from(files));
  } catch (error) {
    console.error(error);
  }

  if (resetOnChange) currentTarget.value = '';
};

// And in FileUploadButton.tsx and ImageUploadButton.tsx
const onFileChange = ({ currentTarget }: React.ChangeEvent<HTMLInputElement>) => {
  const updatedFiles = currentTarget.files;
  handleFileChange(updatedFiles, resetOnChange, handleFiles);
}

<input
  ...
  onChange={onFileChange}
/>
```

Reply:
I see where you're coming from and I'm sorry for confusing function naming. As for your snippet - it makes sense - but I'll go for a hook to make it more React-y and so you wouldn't have to define that function twice. 

```tsx
// utils.ts
export const useHandleFileChangeWrapper = (
  resetOnChange: boolean = false,
  handler?: (files: Array<File>) => void,
) =>
  useCallback(
    ({ currentTarget }: React.ChangeEvent<HTMLInputElement>) => {
      const { files } = currentTarget;

      if (!files) return;

      try {
        handler?.(Array.from(files));
      } catch (error) {
        console.error(error);
      }

      if (resetOnChange) currentTarget.value = '';
    },
    [callback, resetOnChange],
  );

// in <File/Image>UploadButton.tsx then:
const onFileChange = useHandleFileChangeWrapper(resetOnChange, handleFiles);

<input onChange={onFileChange} />;
```
This way - no more unwanted renders and naming is clear.